### PR TITLE
Add method for affiliation analysis

### PR DESCRIPTION
**Pull request by trangdata**

### CODE REVIEW

**Comment by cgreene** [comment]
One of the things we've gotten critiqued over is grouping countries. I don't think the critique undercuts the primary message of the paper, but I think we might get a lot of mileage out of a table that breaks down results by country. This was infeasible with names, but should be feasible with affiliations.

I'm imagining a table with the product of literature fraction * number of honorees compared against the actual number of honorees from that country.

**Comment by dhimmel** [comment]
> I think we might get a lot of mileage out of a table that breaks down results by country. This was infeasible with names, but should be feasible with affiliations.

Yes, I think a country enrichment / depletion analysis would be really helpful and more fine-grained then regions.

However, I it would be best to proceed with this PR as is, and decide to replace the region-affiliation analyses later.

I'll review line-by-line now.

File: content/20.results.md [hunk 1]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
...
+
+![Compared to Pubmed authors, honorees affiliated with institutions from North America are overrepresented while those from Europe and Central Asia, East Asia & Pacific group are underrepresented.
+Estimated composition of affiliation locations over the years of (A, left) all Pubmed computational biology and bioinformatics journal authors, and (A, right) all ISCB Fellows and keynote speakers was computed as the geographical proportions of Pubmed articles or ISCB honorees affiliations each year.
+For each region, the geographical proportions of Pubmed articles affiliations are shown as teal LOESS curves, and the geographical proportions and 95% confidence interval of the ISCB affiliations are shown as dark circles and vertical lines (B).
+](https://raw.githubusercontent.com/greenelab/iscb-diversity/master/figs/region_breakdown.png){#fig:affiliation_breakdown}
```
Comment (dhimmel) [inline content/20.results.md]: What do the letters correspond to in this figure? The country groupings in

- [`2020-01-31_natural-earth-world-map.png`](https://raw.githubusercontent.com/greenelab/iscb-diversity/b41b8fad157d0b878a2476cec998aba0643742bf/figs/2020-01-31_natural-earth-world-map.png)
- the nameprism groupings from Figure 1?

If the former, it's a bit confusing to have letters refer to different groupings. Could we just replace with the standard world geographical mapping region labels?

File: content/10.methods.md [hunk 2]
```diff
@@ -146,3 +146,11 @@ The table lists the 8 groups and the number of living people for each region tha
...
 Full information about which countries comprised each region can be found in the online dataset [`country_to_region.tsv`](https://github.com/greenelab/wiki-nationality-estimate/blob/master/data/country_to_region.tsv).
 {#tbl:example_names}
+
+### Affiliation Analysis
+
+Along with the corresponding author names, we collected their affiliations recorded in each publication for this analysis.
+During the honoree curation process, if an honoree was listed with their affiliation at the time, we recorded this affiliation for analysis.
+Because we could not find affiliations for the 1997 and 1998 RECOMB keynote speakers' listed for these years, they were left blank.
+We used the [standard world geographical mapping](https://raw.githubusercontent.com/greenelab/iscb-diversity/b41b8fad157d0b878a2476cec998aba0643742bf/figs/2020-01-31_natural-earth-world-map.png) from the R package `rnaturalearth` to group the countries of affiliation in seven different regions: North America, Europe & Central Asia, East Asia & Pacific, Latin America & Caribbean, South Asia, Middle East & North Africa and Sub-Saharan Africa.
```
Comment (dhimmel) [inline content/10.methods.md]: I can't find anything online for "standard world geographical mapping". Tracking the data back its from the `region_wb` column of [`data/countries/world-map.tsv`](https://github.com/greenelab/iscb-diversity/blob/96ad84b0c58c3b3de301056f7546664f8bacb057/data/countries/world-map.tsv). Sounds like this is from Natural Earth. I'll look for some documentation of `region_wb` that we can point to... basically who makes it and via what criteria / what `region_wb` is defined as.

File: content/20.results.md (hunk 1 above)
Comment (trangdata) [reply content/20.results.md]: My bad. The linked figure is incorrect. It should be fixed now.

File: content/10.methods.md [hunk 3]
```diff
@@ -146,3 +146,11 @@ The table lists the 8 groups and the number of living people for each region tha
 Example names shows actual author names that received a high prediction for each region.
 Full information about which countries comprised each region can be found in the online dataset [`country_to_region.tsv`](https://github.com/greenelab/wiki-nationality-estimate/blob/master/data/country_to_region.tsv).
 {#tbl:example_names}
+
+### Affiliation Analysis
+
+Along with the corresponding author names, we collected their affiliations recorded in each publication for this analysis.
+During the honoree curation process, if an honoree was listed with their affiliation at the time, we recorded this affiliation for analysis.
```
Comment (cgreene) [inline content/10.methods.md:153]:
```suggestion
During the honoree curation process, if an honoree was listed with their affiliation at the time, we recorded this affiliation for analysis.
For ISCB Fellows, we used the affiliation listed on the ISCB page.
```
This is what I did.

File: content/10.methods.md (hunk 2 above)
Comment (cgreene) [reply content/10.methods.md]: I agree that we need to source the mapping.

File: content/20.results.md [hunk 4]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
   For each region, the mean predicted probability of Pubmed articles is shown as teal LOESS curve, and the mean probability and 95% confidence interval of the ISCB honoree predictions are shown as dark circles and vertical lines (B).
 
 ](https://raw.githubusercontent.com/greenelab/iscb-diversity/master/figs/region_breakdown.png){#fig:region_breakdown}
+
+
+### Assessing Affiliation Diversity of Authors and Honorees
```
Comment (cgreene) [inline content/20.results.md]:
```suggestion
### Assessing Affiliation Diversity of Authors and Honorees

```
I think this needs a newline, unless headers are special.

File: content/20.results.md [hunk 5]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
...
+We observed an increasing proportion of author affiliation located in East Asia & Pacific (Fig. {@fig:affiliation_breakdown}A, left).
```
Comment (cgreene) [inline content/20.results.md]:
```suggestion
We observed an increasing proportion of author affiliations in East Asia & the Pacific (Fig. {@fig:affiliation_breakdown}A, left).
```

File: content/20.results.md [hunk 6]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
...
+This was primarily driven by publications in Bioinformatics and BMC Bioinformatics (see [notebook](https://greenelab.github.io/iscb-diversity/15.analyze-affiliation.html#sup_fig_s6)).
```
Comment (cgreene) [inline content/20.results.md]:
```suggestion
This was primarily driven by publications in _Bioinformatics_ and _BMC Bioinformatics_ (see [notebook](https://greenelab.github.io/iscb-diversity/15.analyze-affiliation.html#sup_fig_s6)).
```
I think we've been going italics

File: content/20.results.md [hunk 7]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
...
+Among ISCB keynote speakers and fellows, we found that the majority of honorees are affiliated with institutions and companies from North America (Fig. {@fig:affiliation_breakdown}A, right).
```
Comment (cgreene) [inline content/20.results.md]:
```suggestion
Among ISCB keynote speakers and fellows, we found that the majority of honorees were affiliated with institutions and companies from North America (Fig. {@fig:affiliation_breakdown}A, right).
```

File: content/20.results.md [hunk 8]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
...
+Noting the small sample size, we found discrepancies between the group of honoree composition versus PubMed.
```
Comment (cgreene) [inline content/20.results.md]:
```suggestion
We observed discrepancies between the affiliations of honorees and article authors.
```

File: content/20.results.md [hunk 9]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
...
 
 ](https://raw.githubusercontent.com/greenelab/iscb-diversity/master/figs/region_breakdown.png){#fig:region_breakdown}
+
+
+### Assessing Affiliation Diversity of Authors and Honorees
+We observed an increasing proportion of author affiliation located in East Asia & Pacific (Fig. {@fig:affiliation_breakdown}A, left).
+This was primarily driven by publications in Bioinformatics and BMC Bioinformatics (see [notebook](https://greenelab.github.io/iscb-diversity/15.analyze-affiliation.html#sup_fig_s6)).
+Among ISCB keynote speakers and fellows, we found that the majority of honorees are affiliated with institutions and companies from North America (Fig. {@fig:affiliation_breakdown}A, right).
+Noting the small sample size, we found discrepancies between the group of honoree composition versus PubMed. 
+Compared to Pubmed authors, ISCB keynote speakers and fellows have a higher proportion of individuals affiliated with North American institutions and companies (Fig. {@fig:affiliation_breakdown}B) and lower proportion of individuals affiliated with institutions or companies from Europe & Central Asia and East Asia & Pacific.
```
Comment (cgreene) [inline content/20.results.md]:
```suggestion
ISCB keynote speakers and fellows had a higher proportion of individuals affiliated with North American institutions and companies (Fig. {@fig:affiliation_breakdown}B) and lower proportion of individuals affiliated with institutions or companies from Europe & Central Asia and East Asia & Pacific than authors. 
```

File: content/20.results.md [hunk 10]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
...
+We observed an increasing proportion of author affiliation located in East Asia & Pacific (Fig. {@fig:affiliation_breakdown}A, left).
+This was primarily driven by publications in Bioinformatics and BMC Bioinformatics (see [notebook](https://greenelab.github.io/iscb-diversity/15.analyze-affiliation.html#sup_fig_s6)).
+Among ISCB keynote speakers and fellows, we found that the majority of honorees are affiliated with institutions and companies from North America (Fig. {@fig:affiliation_breakdown}A, right).
+Noting the small sample size, we found discrepancies between the group of honoree composition versus PubMed. 
+Compared to Pubmed authors, ISCB keynote speakers and fellows have a higher proportion of individuals affiliated with North American institutions and companies (Fig. {@fig:affiliation_breakdown}B) and lower proportion of individuals affiliated with institutions or companies from Europe & Central Asia and East Asia & Pacific. 
+
+![Compared to Pubmed authors, honorees affiliated with institutions from North America are overrepresented while those from Europe and Central Asia, East Asia & Pacific group are underrepresented.
```
Comment (cgreene) [inline content/20.results.md]:
```suggestion
![Compared to authors, honorees affiliated with institutions from North America were overrepresented while those from Europe and Central Asia, East Asia & Pacific group were underrepresented.
```

File: content/20.results.md [hunk 11]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
...
+Estimated composition of affiliation locations over the years of (A, left) all Pubmed computational biology and bioinformatics journal authors, and (A, right) all ISCB Fellows and keynote speakers was computed as the geographical proportions of Pubmed articles or ISCB honorees affiliations each year.
```
Comment (cgreene) [inline content/20.results.md]:
```suggestion
We estimated the composition of affiliation locations each year for (A, left) computational biology and bioinformatics journal authors, and (A, right) all ISCB Fellows and keynote speakers.
```

File: content/20.results.md [hunk 12]
```diff
@@ -98,3 +98,16 @@ Outside of the primary range of our analyses, the two names of 2020 PSB keynote
...
+For each region, the geographical proportions of Pubmed articles affiliations are shown as teal LOESS curves, and the geographical proportions and 95% confidence interval of the ISCB affiliations are shown as dark circles and vertical lines (B).
```
Comment (cgreene) [inline content/20.results.md]:
```suggestion
(B) For each region, we plotted the geographical proportions of article authors as teal LOESS curves, and ISCB honorees are shown as dark circles and vertical lines (error indicators show 95% confidence intervals).
```

**Review by cgreene — COMMENTED** [review]
I had a few suggested changes. I am not confident in the wording of all of them, so please take a careful look :)

File: content/20.results.md (hunk 6 above)
Comment (trangdata) [reply content/20.results.md]: ah yes sorry it's been a while

File: content/20.results.md (hunk 9 above)
Comment (trangdata) [reply content/20.results.md]: I left "Compared to authors" at the beginning of this sentence for more direct reading. I fixed the tense as you suggested.

**Comment by trangdata** [comment]
> One of the things we've gotten critiqued over is grouping countries. I don't think the critique undercuts the primary message of the paper, but I think we might get a lot of mileage out of a table that breaks down results by country. This was infeasible with names, but should be feasible with affiliations.

[Affiliation analysis html](https://greenelab.github.io/iscb-diversity/15.analyze-affiliation.html) now updated by [this commit](https://github.com/greenelab/iscb-diversity/commit/821fde0d35c291e3878a2986fc66bb3057439e7b).

File: content/20.results.md (hunk 9 above)
Comment (cgreene) [reply content/20.results.md]: 👍

**Comment by trangdata** [comment]
Now that the country-level analysis is in much better shape, should we replace the group level analysis with it? @cgreene @dhimmel

**Comment by cgreene** [comment]
I think that would be a good idea. I feel like the country level analysis addresses a lot of the concerns that have been raised by Iddo in #27 and others.

**Comment by trangdata** [comment]
OK I'm happy to close this issue and open a new one if that's easiest for everyone! Or we can merge to keep part of the Methods section and I'll make another PR to change what we want to focus on in the Results.

**Comment by cgreene** [comment]
Well, the methods changes would apply regardless except for the bit about grouping. What do you think about keeping those changes and dropping the changes to the results. The results changes with a country-level analysis could be added by a separate pull request. We could then go ahead and merge this if we wanted to (I'm more comfortable merging methods before results than vice versa) or hold on until they're both ready and merge about the same time.

**Review by cgreene — APPROVED** [review]
Looks good!

File: content/10.methods.md [hunk 13]
```diff
@@ -146,3 +146,12 @@ The table lists the 8 groups and the number of living people for each region tha
...
+For ISCB Fellows, we used the affiliation listed on the ISCB page.
+Because we could not find affiliations for the 1997 and 1998 RECOMB keynote speakers' listed for these years, they were left blank.
+We used the [World Bank analytical grouping](https://raw.githubusercontent.com/greenelab/iscb-diversity/b41b8fad157d0b878a2476cec998aba0643742bf/figs/2020-01-31_natural-earth-world-map.png) implemented in the R package `rnaturalearth` to group the countries of affiliation in seven different regions: North America, Europe & Central Asia, East Asia & Pacific, Latin America & Caribbean, South Asia, Middle East & North Africa and Sub-Saharan Africa.
```
Comment (cgreene) [inline content/10.methods.md]: Oh - wait - we should probably remove this line.

Comment (cgreene) [inline content/10.methods.md]:
```suggestion
```

**Pull request merged**